File: func/recur/fibs.erl
% Enter your code here. Read input from STDIN. Print output to STDOUT
% Your class should be named solution

-module(solution).
-export([main/0]).

main() ->
	{ok,[X]} = io:fread("","~d"),
    io:format("~w~n",[fib(X)]).

fib(1) ->
    0;
fib(2) ->
    1;
fib(N) ->
    fib(N - 3, 0, 1).

fib(0, X, Y) ->
    X + Y;
fib(N, X, Y) ->
    fib(N - 1, Y, (X + Y)).

File: func/recur/fibs.scala
object Solution {
    
    def fibonacci(x:Int):Int = {
        def loop (x: Int, y: Int, n: Int): Int = n match {
            case 0 => x + y
            case i => loop(y, x + y, i - 1)
        }
        x match {
         case 1 => 0
         case 2 => 1
         case _ => loop(0, 1, x - 3)
        }
    }

    def main(args: Array[String]) {
         /** This will handle the input and output**/
         println(fibonacci(readInt()))

    }
}
